Source/WTF/wtf/text/ParsingUtilities.h

```diff
@@ -255,6 +290,37 @@ template<typename CharacterType, std::size_t Extent> constexpr bool skipCharacte
     return true;
 }
 
-} // namespace WebCore
+template<typename T> T& consumeSingleElement(std::span<T>& data)
+{
+    T& value = data[0];
+    data = data.subspan(1);
+    return value;
+}
+
+template<typename T> std::span<T> consumeSpan(std::span<T>& data, size_t amountToConsume)
```

Comment:
@darinadler Let me know if you'd prefer if this function (and `consumeSingleElement`) do a length check. Or maybe we'd add another `tryConsumeSpan` that does a length check. For now I kept as is.